src/connector/src/base.rs

```diff
@@ -47,20 +47,36 @@ pub struct SourceMessage {
 }
 
 /// The metadata of a split.
-pub trait SourceSplit: Sized {
+pub trait SplitMetaData: Sized {
     fn id(&self) -> String;
-    fn to_string(&self) -> Result<String>;
+    fn to_json_bytes(&self) -> Result<Bytes>;
     fn restore_from_bytes(bytes: &[u8]) -> Result<Self>;
 }
 
 /// The persistent state of the connector.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ConnectorState {
     pub identifier: Bytes,
     pub start_offset: String,
     pub end_offset: String,
 }
 
+impl SplitMetaData for ConnectorState {
```

Comment:
I think ConnectorState is not a SplitMetaData? It's not going to be used by `list_splits()`.

```suggestion
impl ConnectorState {
```

Reply:
The contents of the `SourceStateHandler` store need to implement the trait.